src/modules/gravityCompensator/main.cpp

```diff
@@ -227,12 +227,31 @@ class gravityModuleCompensator: public RFModule
 
         //------------SPECIAL PARAM TP DEFINE THE HEAD TYPE-----//
         version_tag icub_type;
-        if (rf.check("headV2"))
+        // set icub_type.head_version int to 1 (default) or 2
+        if (rf.check("head_version"))
         {
-            yInfo("'headV2' option found. Using icubV2 head kinematics.\n");
-            icub_type.head_version = 2;
+            if (rf.find("head_version").asDouble()==1.0)
+            {
+                yInfo("\"head_version\" %g requested => Using V1 head kinematics.\n",
+                      rf.find("head_version").asDouble());
+            }
+            else if (rf.find("head_version").asDouble()==2.0 ||
+                rf.find("head_version").asDouble()==2.5)
+            {
+                yInfo("\"head_version\" %g requested => Using V2 head kinematics.\n",
+                      rf.find("head_version").asDouble());
+                icub_type.head_version = 2;
```

Comment:
Yes, the PR is cosmetic, in the sense that it was only focused on user-friendliness: having `--head_version` everywhere instead of `--head_version` for some modules, `--headV2` for other ones.

When I looked at the code and wrote the parsing functions, I also realized the problem that "2.5 should not be treated as 2.0 kinematics" (which was not introduced by this PR although it was made more evident by it), but I did not feel comfortable or qualified to touch `iDyn`. :)

By all means, if you prefer to upgrade everything together in the future in a consistent manner (both the syntax and the kinematics instantiation), close this PR and follow the more robust approach.

Otherwise, if you prefer to go step by step, let me know if I should push any more changes here.

In any case, sorry for not being clear here or in https://github.com/robotology/icub-main/issues/405 which (for me) originated this PR :)